### PR TITLE
Fix /users patch endpoint, add more fields to model

### LINKED ISSUE #9: /users endpoints
Update database model to include all fields in the create user form.

https://www.figma.com/design/j9QIHvgM0wOVw3xDe9KdWB/My-Copy%3A-Cohort-Manager-app---Designs?node-id=228-41540&t=ep6tuwERh4cTL9t4-0

## Model changes
- Add field `mobile` to `User` class
- Add field `specialism` to `User` class
- Add field `imageUrl` to `User` class

## Add deleteById endpoint
Method: DELETE
Url: /users/{id}
Response body type: `User`
Error conditions:
- No user with `id`:
    - Response body:
```json
{
  "status": "fail",
  "message": "User not found"
}
```

## Fix /users PATCH endpoint
Existing implementation of PATCH endpoint only consisted of hardcoded responses and did not communicate with the database. Now it stores patched changes to the given user persistently

Method: PATCH
Url: /users/{id}
Response body type: `User`

### PR DESCRIPTION
**Pull request by gremble0**
- Add more fields to the model and database.
- Update openapi documentation.
- Be more consistent with returns in user controller.

### CODE REVIEW

**Review by JosteinRuen — DISMISSED** [review]
Amazing work

**Review by JosteinRuen — DISMISSED** [review]
Great job updating the model

**Review by thomaafl — DISMISSED** [review]
lovely code

File: docs/openapi.yml [hunk 1]
```diff
@@ -164,6 +164,34 @@ paths:
...
+          description: Unique identifier of the user
+          schema:
+            type: integer
+      responses:
+        '201':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CreatedUser'
```
Comment (allendewilzin) [inline docs/openapi.yml]: why is the response createdUser model. 

user wont exist anymore

**Review by allendewilzin — CHANGES_REQUESTED** [review]

File: docs/openapi.yml (hunk 1 above)
Comment (gremble0) [reply docs/openapi.yml]: The model for a created user and a deleted user is the same so no need to duplicate. Renamed `CreatedUser` to `User` to better reflect

**Review by allendewilzin — APPROVED** [review]

**Comment by allendewilzin** [comment]
Good code.

**Review by thomaafl — APPROVED** [review]

**Pull request merged**